libs/platform/src/lib/table-helpers/directives/table-advanced-viewport.directive.ts

```diff
@@ -0,0 +1,128 @@
+import {
+    Directive,
+    ElementRef,
+    Input,
+    Renderer2,
+    OnInit,
+    TemplateRef,
+    ViewContainerRef,
+    ChangeDetectorRef,
+    inject,
+    OnDestroy
+} from '@angular/core';
+import { Observable, take, takeUntil, Subscriber, Subject } from 'rxjs';
+import { DestroyedService } from '@fundamental-ngx/cdk/utils';
+import { ViewportRootService } from '../services/table-advanced-viewport-root.service';
+
+@Directive({
+    selector: '[fdpTableViewport]',
+    standalone: true,
+    providers: [DestroyedService]
+})
+export class ViewportDirective implements OnInit, OnDestroy {
```

Comment:
Need a more specific name here and for the filename as well. It should give us an idea as to what the directive does

Reply:
Updated